ambry-router/src/test/java/com.github.ambry.router/NonBlockingRouterTest.java

```diff
@@ -74,9 +73,19 @@ private void setOperationParams() {
   @Test
   public void testNonBlockingRouterFactory()
       throws Exception {
-    Properties props = getNonBlockingRouterProperties();
-    VerifiableProperties verifiableProperties = new VerifiableProperties((props));
     MockClusterMap mockClusterMap = new MockClusterMap();
+    try {
+      Properties props = getNonBlockingRouterProperties("NotInClusterMap");
+      VerifiableProperties verifiableProperties = new VerifiableProperties((props));
+      router = (NonBlockingRouter) new NonBlockingRouterFactory(verifiableProperties, mockClusterMap,
+          new LoggingNotificationSystem()).getRouter();
+      Assert.fail("NonBlockingRouterFactory instantiation should have failed because the router datacenter is not in "
```

Comment:
could you please keep _only_ that statement that throws the exception and the `fail()` in the `try` block?